rainier-core/src/main/scala/com/stripe/rainier/core/Injection.scala

```diff
@@ -17,21 +17,35 @@ trait Injection { self =>
     See https://en.wikipedia.org/wiki/Probability_density_function#Dependent_variables_and_change_of_variables
     This function should be log(d/dy backwards(y)), where y = forwards(x).
    */
-  def logJacobian(y: Real): Real
+  def logBackwardsJacobian(y: Real): Real
+  def logForwardsJacobian(x: Real): Real
 
-  def transform(dist: Continuous): Continuous = new Continuous {
-    def realLogDensity(real: Real): Real =
-      If(isDefinedAt(real),
-         dist.realLogDensity(backwards(real)) +
-           logJacobian(real),
-         Real.zero.log)
+  def transform(dist: Continuous): Continuous = {
+    val oldSupport = dist.support
 
-    val generator: Generator[Double] =
-      Generator.require(self.requirements) { (r, n) =>
-        n.toDouble(forwards(dist.generator.get(r, n)))
-      }
+    val newSupport = new Support {
+      def transform(v: Variable): Real = forwards(oldSupport.transform(v))
 
-    def param: RandomVariable[Real] = dist.param.map(forwards)
+      def logJacobian(v: Variable): Real =
+        oldSupport.logJacobian(v) + logForwardsJacobian(oldSupport.transform(v))
+
+      def isDefinedAt(y: Real): Real = oldSupport.isDefinedAt(backwards(y))
+    }
+
+    new Continuous {
+      def realLogDensity(real: Real): Real =
+        If(newSupport.isDefinedAt(real),
```

Comment:
This `If` statement is causing the "monad laws with all reals" test to fail. TODO look into it.